src/main/java/hudson/plugins/parameterizedtrigger/NodeAction.java

```diff
@@ -34,23 +34,29 @@
 import java.util.List;
 
 /**
- * {@link Action} that restricts the job to a particular node 
+ * {@link Action} that restricts the job to a particular node
  * when a project is scheduled
  * Will cause a unique build for each different node if a job is already queued.
- * 
+ *
  * @author Chris Johnson
  */
 public class NodeAction extends InvisibleAction implements LabelAssignmentAction, Queue.QueueAction, BuildBadgeAction {
 	private final Label nodeLabel;
-	
+
 	public NodeAction(Label nodeLabel) {
 		this.nodeLabel = nodeLabel;
 	}
 
 	public Label getAssignedLabel(SubTask task) {
-		return nodeLabel;
+		if (nodeLabel != null) {
```

Comment:
I guess this is in keeping with the rest of the file, but man, 8 spaces for indenting??

Reply:
In this repo it's real tabs I think (`\t`).  Not a fan but luckily VSCode is smart enough that I didn't even notice while editing.